### DISCUSSION TITLE
3.x+: A newbie talk something about queue, task status.

### DISCUSSION THREAD

**Discussion by NItaMII** [Netflix/conductor#3251] [Show and tell]
My situation: the workflow instance need to go continue when some signals introduced. For example, I add a Wait Task before a Simple Task (some business logical), then this flow will be stop there until someone trigger a signal. 

- The 'Event Handler' (not fits my case.)
- Update Task status by API 'queue/update/{workflowId}/{taskRefName}/{status}'

I try the api, but get ERROR such like 'There is no Completed Queue'. Then I find so many issues （someone like me, is new to Conductor）, asking topics about 'How to create event queue' or 'why update task throw an error'. 

So, for these newbies(including me), maybe they just want to using ConductorEventQueue rather than 'sqs', 'amqp'. Just as simple as possible, but I can't find docs about ConductorEventQueue.

```
    @ConditionalOnProperty(name = "conductor.default-event-queue.type", havingValue = "conductor")
    @Bean
    public Map<TaskModel.Status, ObservableQueue> getQueues(
            ConductorProperties conductorProperties, EventQueueProvider provider) {
        String stack = "";
        if (conductorProperties.getStack() != null && conductorProperties.getStack().length() > 0) {
            stack = conductorProperties.getStack() + "_";
        }
        TaskModel.Status[] statuses =
                new TaskModel.Status[] {TaskModel.Status.COMPLETED, TaskModel.Status.FAILED};
        Map<TaskModel.Status, ObservableQueue> queues = new HashMap<>();

        for (TaskModel.Status status : statuses) {
            String queuePrefix = conductorProperties.getAppId() + "_default_notify_" + stack;
            String queueName = queuePrefix + status.name();

            ObservableQueue queue = provider.getQueue(queueName);
            queues.put(status, queue);
        }

        return queues;
    }
```

**Comment by apanicker-nflx** [comment]
@NItaMII The event queues as per status are provided by the SQS implementation by default - https://github.com/Netflix/conductor/blob/main/awssqs-event-queue/src/main/java/com/netflix/conductor/sqs/config/SQSEventQueueConfiguration.java#L37

This configuration provides event queues and listeners per status. You can follow the same pattern and use the bean you provided as an example to generate a ConductorEventQueue within your implementation. 
The caveat being that instead of relying on EventQueueProvider for getting the event queue implementation, I would recommend using the queue implementation that is used for task queues to be re-used for event queues.
